build/android.mk

```diff
@@ -202,75 +189,48 @@ $(ANDROID_BUILD)/classes.dex: $(JAVA_SOURCES) $(ANDROID_BUILD)/gen/org/xcsoar/R.
-	$$(Q)$$(MAKE) TARGET=$(3) DEBUG=$$(DEBUG) USE_CCACHE=$$(USE_CCACHE) $$@
+# Run the sub-makes every time but ensure that the native headers are generate (happens with the build of the DEX file.
+$$(TARGET_OUTPUT_DIR)/$$(ANDROID_APK_LIB_ABI)/$$(XCSOAR_ABI)/bin/lib$(1).so: FORCE $$(ANDROID_BUILD)/classes.dex
+	$$(Q)$$(MAKE) TARGET=$(2) ANDROID_BUILD=$$(ANDROID_BUILD) NATIVE_INCLUDE=$$(NATIVE_INCLUDE) ANDROID_BUILD_APK=n DEBUG=$$(DEBUG) USE_CCACHE=$$(USE_CCACHE) $$@
 
 endef
 
-# Example: $(eval $(call generate-abi,xcsoar))
+# Example: $(eval $(call generate-all-abis,xcsoar))
 define generate-all-abis
-$(eval $(call generate-abi,$(1),armeabi-v7a,ANDROID7))
-$(eval $(call generate-abi,$(1),x86,ANDROID86))
-$(eval $(call generate-abi,$(1),arm64-v8a,ANDROIDAARCH64))
-$(eval $(call generate-abi,$(1),x86_64,ANDROIDX64))
+$(foreach T,$(ANDROID_TARGETS),$(call generate-abi,$(1),$(T)))
 endef
+#$(foreach T,$(ANDROID_TARGETS),$(eval $(call generate-abi,$(ANDROID_LIB_NAMES),$(T))))
 
 $(foreach NAME,$(ANDROID_LIB_NAMES),$(eval $(call generate-all-abis,$(NAME))))
 
-.PHONY: libs
-libs: $(ANDROID_THIRDPARTY_STAMPS)
-
-else # !FAT_BINARY
-
-# add dependency to this source file
-$(call SRC_TO_OBJ,$(SRC)/Android/Main.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/EventBridge.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/InternalSensors.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/Battery.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/GliderLink.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/NativePortListener.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/NativeLeScanCallback.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/NativeInputListener.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/DownloadManager.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/NativeBMP085Listener.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/NativeI2CbaroListener.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/NativeNunchuckListener.cpp): $(NATIVE_HEADERS)
-$(call SRC_TO_OBJ,$(SRC)/Android/NativeVoltageListener.cpp): $(NATIVE_HEADERS)
-
-ANDROID_LIB_BUILD = $(patsubst %,$(ANDROID_ABI_DIR)/lib%.so,$(ANDROID_LIB_NAMES))
-$(ANDROID_LIB_BUILD): $(ANDROID_ABI_DIR)/lib%.so: $(ABI_BIN_DIR)/lib%.so | $(ANDROID_ABI_DIR)/dirstamp
-	$(Q)cp $< $@
-
-endif # !FAT_BINARY
-
-
-$(NATIVE_HEADERS): $(ANDROID_BUILD)/classes.dex
-
 .DELETE_ON_ERROR: $(ANDROID_BUILD)/unsigned.apk
-$(ANDROID_BUILD)/unsigned.apk: $(ANDROID_BUILD)/classes.dex $(ANDROID_BUILD)/resources.apk $(ANDROID_LIB_BUILD)
+$(ANDROID_BUILD)/unsigned.apk: cleanNativeLibs $(ANDROID_BUILD)/classes.dex $(ANDROID_BUILD)/resources.apk $(ANDROID_LIB_BUILD)
 	@$(NQ)echo "  APK     $@"
 	$(Q)cp $(ANDROID_BUILD)/resources.apk $@
 	$(Q)cd $(dir $@) && zip -q -r $(notdir $@) classes.dex lib
 
+# Remove potentially stale native shared libs before building and copying new ones to the lib directory
```

Comment:
What does this mean?

Reply:
Better word is outdated. I re-wrote the comment to make the purpose clear. Please see the updated source.